src/test/java/com/google/api/gax/grpc/RecordingScheduler.java

```diff
@@ -29,41 +29,63 @@
  */
 package com.google.api.gax.grpc;
 
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.when;
+
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 import org.joda.time.Duration;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
 
-class RecordingScheduler implements UnaryCallable.Scheduler {
-  private final ScheduledExecutorService executor;
-  private final List<Duration> sleepDurations = new ArrayList<>();
-  private final FakeNanoClock clock;
+abstract class RecordingScheduler implements ScheduledExecutorService {
 
-  public RecordingScheduler(FakeNanoClock clock) {
-    this.executor = new ScheduledThreadPoolExecutor(1);
-    this.clock = clock;
-  }
+  abstract List<Duration> getSleepDurations();
 
-  @Override
-  public ScheduledFuture<?> schedule(Runnable runnable, long delay, TimeUnit unit) {
-    sleepDurations.add(new Duration(TimeUnit.MILLISECONDS.convert(delay, unit)));
-    clock.setCurrentNanoTime(clock.nanoTime() + TimeUnit.NANOSECONDS.convert(delay, unit));
-    return executor.schedule(runnable, 0, TimeUnit.NANOSECONDS);
-  }
+  static RecordingScheduler get(final FakeNanoClock clock) {
```

Comment:
maybe call this `create` instead of `get` - it matches other classes in GAX better.